src/core/rent_collector.zig

```diff
@@ -127,8 +127,10 @@ pub const RentCollector = struct {
     ) RentDue {
         if (self.rent.isExempt(lamports, data_len)) return .Exempt;
 
+        if (account_rent_epoch > self.epoch) return .{ .Paying = 0 };
+
         var slots_elapsed: u64 = 0;
-        for (account_rent_epoch..self.epoch + 1) |epoch| {
+        for (account_rent_epoch..self.epoch +| 1) |epoch| {
```

Comment:
Is the saturating arithmetic correct/needed here? I think the overflow that was being hit here was only due to `account_rent_epoch > self.epoch + 1`. If this happens to overflow, then I think that's a very different failure state which shouldn't be hidden by this saturating op.